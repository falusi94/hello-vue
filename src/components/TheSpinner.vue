<template>
  <div>
    <div class="spinner-container">
      <img class="spinner-image" src="../assets/spinner.png" alt="Loading"/>
    </div>
    <div class="spinner-text">
      Loading...
    </div>
  </div>
</template>

<script>
export default {
  name: 'TheSpinner',
};
</script>

<style lang="scss" scoped>
@keyframes loading {
  0%   { transform: rotate(0deg);   }
  100% { transform: rotate(360deg); }
}
.spinner {
  &-container {
    margin: 0 auto 0 auto;
    width: 50px;
  }
  &-image {
    animation: loading 1.8s infinite linear;
    width: 50px;
  }
  &-text {
    text-align: center;
    padding: 25px 0;
  }
}
</style>
