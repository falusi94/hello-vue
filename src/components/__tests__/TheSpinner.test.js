import { shallowMount } from '@vue/test-utils';
import TheSpinner from '../TheSpinner.vue';

describe('TheSpinner.vue', () => {
  it('renders properly', () => {
    const wrapper = shallowMount(TheSpinner);
    expect(wrapper).toMatchSnapshot();
  });
});
