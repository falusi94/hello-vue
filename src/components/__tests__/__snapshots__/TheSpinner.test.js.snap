// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheSpinner.vue renders properly 1`] = `
<div>
  <div class="spinner-container"><img src="../assets/spinner.png" alt="Loading" class="spinner-image"></div>
  <div class="spinner-text">
    Loading...
  </div>
</div>
`;
